pkg/api/resource_helpers_test.go

```diff
@@ -118,3 +118,111 @@ func TestIsPodAvailable(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdatePodCondition(t *testing.T) {
+	tests := []struct {
+		podStatus    *PodStatus
+		podCondition *PodCondition
+		expected     bool
+	}{
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodInitialized,
+						Status: ConditionFalse,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: true,
+		},
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodReady,
+						Status: ConditionFalse,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: true,
+		},
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodReady,
+						Status: ConditionTrue,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: false,
+		},
+	}
+
+	for i, test := range tests {
+		changedOrAadd := UpdatePodCondition(test.podStatus, test.podCondition)
+		if changedOrAadd != test.expected {
```

Comment:
would probably be good to test the pod status looks as expected after this is called